File: amic-platform/src/modules/ma/types/workflow.ts
import type { TransactionPhase, TransactionStatus } from "./transaction";

export interface PhaseConfig {
  phase: TransactionPhase;
  label: string;
  description: string;
  icon: string;
  order: number;
}

export interface PhasePrerequisite {
  field: string;
  label: string;
  satisfied: boolean;
  current_value?: string | null;
  target_value?: string | null;
  requires_acknowledgement?: boolean;
}

export interface PhaseCompletionStatus {
  current_phase: TransactionPhase;
  prerequisites: PhasePrerequisite[];
  all_met: boolean;
  can_advance: boolean;
  blocking_reasons: string[];
  next_phase: TransactionPhase | null;
  previous_phase: TransactionPhase | null;
  gate_summary?: string | null;
  pending_acknowledgements: string[];
  requires_user_acknowledgement: boolean;
}

export interface WorkflowTransition {
  from_phase: TransactionPhase;
  to_phase: TransactionPhase;
  transitioned_at: string;
  transitioned_by: string;
  notes: string | null;
}

export interface StatusChange {
  from_status: TransactionStatus;
  to_status: TransactionStatus;
  reason: string | null;
}

File: deal-mgmt/app/schemas/workflow.py
from __future__ import annotations

from pydantic import BaseModel

from app.models.enums import TransactionPhase, TransactionStatus


class PhasePrerequisite(BaseModel):
    field: str
    label: str
    satisfied: bool


class PhaseCompletionStatus(BaseModel):
    current_phase: TransactionPhase
    prerequisites: list[PhasePrerequisite]
    all_met: bool
    can_advance: bool
    blocking_reasons: list[str] = []
    next_phase: TransactionPhase | None = None
    previous_phase: TransactionPhase | None = None


class PhaseTransitionRequest(BaseModel):
    to_phase: TransactionPhase
    notes: str | None = None


class PhaseTransitionResponse(BaseModel):
    from_phase: TransactionPhase
    to_phase: TransactionPhase
    success: bool
    message: str


class StatusChangeRequest(BaseModel):
    to_status: TransactionStatus
    reason: str | None = None


class StatusChangeResponse(BaseModel):
    from_status: TransactionStatus
    to_status: TransactionStatus
    success: bool
    message: str
